tests/testsuite/build_script.rs

```diff
@@ -5508,6 +5508,47 @@ for more information about build script outputs.
         .run();
 }
 
+#[cargo_test]
+fn test_new_syntax_with_old_msrv_and_reserved_prefix() {
```

Comment:
Generally, we recommend splitting up PRs into at least two commits
- Adding of tests (must pass)
- Changing of behavior with requisite test updates

This makes it easier to review how the behavior changed and helps test the test.  A little less important in this case but would still be nice if you have to update this PR

Reply:
Done.